### PR TITLE
Update prereq information in docs

### PR DESCRIPTION
**Pull request by markxnelson**
This PR updates the documentation to make our prereqs and supported platforms clearer.
Specifically, it calls out that we do not support dev/test options like kind, minikube,minishift.

### CODE REVIEW

File: docs-source/content/userguide/introduction/introduction.md [hunk 1]
```diff
@@ -35,6 +35,23 @@ Operator 2.0.1+ is certified for use on OpenShift 3.11.43+, with Kubernetes 1.11
 When using the operator in OpenShift, the `anyuid` security context constraint is required to ensure that WebLogic containers run with a UNIX UID that has the correct permissions on the domain filesystem.
 For more information, see [OpenShift]({{<relref "/security/openshift.md">}}) in the Security section.
 
+### Important note about development-focused Kubernetes distributions
+
+There are a number of development-focused distributions of Kubernetes, like kind, Minikube, Minishift and so on.
+Often these run Kubernetes in a virtual machine on you development machine.  We have found that these distributions
```
Comment (rosemarymarano) [inline docs-source/content/userguide/introduction/introduction.md]: one typo, on you development machine -> on your development machine

File: docs-source/content/userguide/introduction/introduction.md [hunk 2]
```diff
@@ -23,18 +23,49 @@ Detailed instructions are available [here]({{< relref "/userguide/managing-opera
 * Oracle WebLogic Server 12.2.1.3.0 with patch 29135930.
```
Comment (mriccell) [inline docs-source/content/userguide/introduction/introduction.md]: We should mention that Oracle WebLogic versions 12.2.1.4 and later do not require a patch.

Comment (markxnelson) [reply docs-source/content/userguide/introduction/introduction.md]: have we published a 12.2.1.4 image yet?

Comment (markxnelson) [reply docs-source/content/userguide/introduction/introduction.md]: I don't see one, or any samples in the docker-images project - I would rather we fixed that first

Comment (markxnelson) [reply docs-source/content/userguide/introduction/introduction.md]: 12.2.1.4 content can be added in another MR

File: docs-source/content/userguide/introduction/introduction.md [hunk 3]
```diff
@@ -23,18 +23,49 @@ Detailed instructions are available [here]({{< relref "/userguide/managing-opera
...
    * The existing WebLogic Docker image, `container-registry.oracle.com/middleware/weblogic:12.2.1.3 `,
```
Comment (mriccell) [inline docs-source/content/userguide/introduction/introduction.md]: there is an extra space after 12.2.1.3 and comma ","

Comment (rosemarymarano) [reply docs-source/content/userguide/introduction/introduction.md]: I think that it appears ok in the docs. The back tick confers code font and then the comma follows.
https://oracle.github.io/weblogic-kubernetes-operator/userguide/introduction/introduction/

**Pull request merged**